controllers/dspipeline_controller.go

```diff
@@ -501,6 +507,46 @@ func (r *DSPAReconciler) PublishMetrics(dspa *dspav1alpha1.DataSciencePipelinesA
 	}
 }
 
+func (r *DSPAReconciler) GetComponents(ctx context.Context, dspa *dspav1alpha1.DataSciencePipelinesApplication) dspav1alpha1.ComponentStatus {
```

Comment:
Some error handling around the network calls to the API Server and MLMD Envoy could be added here.